module-4/backend/week_two.md

```diff
@@ -0,0 +1,43 @@
+## Week Two - Module 4 Recap
+
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+
+Note: When you're done, submit a PR. 
+
+1. What is Webpack and why is it useful?
+  - Webpack is a build tool that helps make a developer's life easier by bundling dependencies, compressing JS into one file, and making it easier to work with the browser.
+  
+2. When do you want to use event delegation?
+  - When there's a chance of events bubbling up
+  
+3. What's one difference between ES5 and ES6?
+  - Using the backticks to interpolate instead of concatenation in ES5
+  
+4. How are you using the MVC design pattern in your Quantified Self project?
+  - We have broken out the routes into their own files, sort of like controllers, and the models into their own files, and soon we'll have views.
+  
+5. How do you execute raw SQL in node?
+  - You use Knex
+  
+6. What's a callback function and what are some reasons when we use/need callback functions?
+  - A function passed in as an argument in another function that will be executed later. 
+
+7. What is CORS?
+  - Cross-origin resource sharing
+  
+8. What are some steps to avoid CORS?
+  - I really don't understand this well enough to give a good answer, but seems like using simple HTTP and content types is a good place to start...
+
+#### Review  
+
+9. Why do people say "HTTP is stateless"?
+  - Each new request and response has no knowledge of the previous; each one is new
+  
+10. What is a RESTful API?
+  - One that sticks with conventional HTTP verbs (GET, POST, PUT/PATCH, DELETE) in order to facilitate communication between a client and server and conduct CRUD operations
```

Comment:
conventional verbs matched up with their meaning matched up with patterns for the path or url 👍